src/api/earmark.ts

```diff
@@ -22,3 +23,17 @@ export const earmarkGasPrice = (gasPrice: string, userPrint: string): string =>
 
 // simple concatenation, with '0x' for empty data to have `0x<userPrint>` at the least
 export const earmarkTxData = (data = '0x', userPrint: string): string => data + userPrint
+
+const GAS_PER_DATA_CHAR = 8 // wei
+
+// increases pre-estimated gas by the amount needed to save extra earmark data to chain
+// 16 wei per 2 characters
+export const calcEarmarkedGas = (gas: string | number, userPrint: string): string => {
```

Comment:
nit: I would have make this function responsibility only to get the gas for earmark, also I understand that can be static. The earmark size is not changing

Then the SUM of the two gas would happen in the middleware

Reply:
Earmark size doesn't change in our particular use case, because we have a fixed number of flags. But for a generic use case it is nice to make this a function of both gas and userPrint, always

I think the responsibility of both gas and data lies with whatever changes any one of them, walletApi in this case
That way middleware doesn't have to bother with hex->num->hex conversion and is more opaque that way.

Reply:
It was a nit, so it's not a big deal but I'm referring to two things here:
- https://github.com/gnosis/dex-react/pull/1445/files#diff-515503171f912f8f40997a12b3f94afcR371 This line is modifying the gas price, adding some "extra". I think this "getting extra" should be independent from the calculation on the result amount. For me that function doesn't need the "gas" param. What you do with that extra gas is the middleware responsability imo (I'm if we leave this untouched, this is a ubernit)
- Once you have a function that returns the extra gas for the 8 hex digits, you don't need to calculate it over and over for every call, since the earmark data doesn't change. It could be statically calculated or cached. `userPrint` don't change. 

Neither of the above are really important

Reply:
> Once you have a function that returns the extra gas for the 8 hex digits, you don't need to calculate it over and over for every call, since the earmark data doesn't change. It could be statically calculated or cached. userPrint don't change.

You convinced me. Will change, on monday. `walletApi.userPrint = {dataSuffix: string, extraGas: number}` or something